File: c/LeetCode_2937_MakeThreeStringsEqual.c
#include <stdio.h>


int findMinimumOperations(char* s1, char* s2, char* s3) {
    int l1 = strlen(s1);
    int l2 = strlen(s2);
    int l3 = strlen(s3);
    int min = (l1 < l2) ? l1 : l2;
    min = (l3 < min) ? l3 : min;

    int counter = 0;
    for (int i = 0; i < min; ++i) {
        if (s1[i] == s2[i] && s1[i] == s3[i]) {
            ++counter;
        } else {
            break;
        }
    }

    if (counter == 0) { return -1; }

    return (l1 + l2 + l3) - (counter * 3);
}

File: python/LeetCode_2937_MakeThreeStringsEqual.py
class Solution:
    def findMinimumOperations(self, s1: str, s2: str, s3: str) -> int:
        low = min(min(len(s1), len(s2)), len(s3))

        prefix = ''
        for i in range(low):
            if (s1[i] == s2[i] == s3[i]):
                prefix += s1[i]
            else:
                break

        if len(prefix) == 0: return -1
        return (len(s1) + len(s2) + len(s3)) - len(prefix) * 3